src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -117,8 +120,8 @@ private BoundExpression VisitIndexerAccess(BoundIndexerAccess node, bool isLeftO
                 // This node will be rewritten with MakePropertyAssignment when rewriting the enclosing BoundAssignmentOperator.
 
                 return oldNodeOpt != null ?
-                    oldNodeOpt.Update(rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, argsToParamsOpt, defaultArguments, type) :
-                    new BoundIndexerAccess(syntax, rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, argsToParamsOpt, defaultArguments, type);
+                    oldNodeOpt.Update(rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, AccessorKind.Set, argsToParamsOpt, defaultArguments, type) :
+                    new BoundIndexerAccess(syntax, rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, AccessorKind.Set, argsToParamsOpt, defaultArguments, type);
```

Comment:
It is not obvious whether ```AccessorKind.Set``` is always an accurate value to use here. For example, are we going to get here for an assignment to a ref returning indexer in an object initializer?

Reply:
This path won't be reached for a ref returning indexer cause the `if` on line 114 blocks that.

Overall it seemed that since this code path is (LHS of assignment + RefKind.None) it should be okay to assume `AccessorKind.Set` here. Please let me know if I'm missing some consideration here. 

Have gone through a number of examples locally to make sure the logic lines up here. There are a few that we're not covering in the ref tests that Im' going to add for completeness.

Reply:
> This path won't be reached for a ref returning indexer cause the if on line 114 blocks that.

Indeed. I missed that because GitHub collapsed code in unchanged region.